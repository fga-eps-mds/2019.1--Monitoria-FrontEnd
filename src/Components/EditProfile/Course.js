import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import OutlinedInput from '@material-ui/core/OutlinedInput';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
  },
  selectEmpty: {
    marginTop: theme.spacing.unit * 2,
  },
});

class SimpleSelect extends React.Component {
  state = {
    selectedCourse:0,
    labelWidth: 0,
  };

  componentDidMount() {
    this.setState({
      labelWidth: ReactDOM.findDOMNode(this.InputLabelRef).offsetWidth,
    });
  }

  handleChange = event => {
    this.setState({ selectedCourse: event.target.value });
  };

  render() {
    const { classes } = this.props;

    return (
      <form className={classes.root} autoComplete="off">
        
        <FormControl variant="outlined" className={classes.formControl}>
          <InputLabel
            ref={ref => {
              this.InputLabelRef = ref;
            }}
            htmlFor="outlined-age-simple"
          >
            Curso
          </InputLabel>
          <Select
            value={this.state.selectedCourse}
            onChange={this.handleChange}
            input={
              <OutlinedInput
                labelWidth={this.state.labelWidth}
                id="course"
              />
            }
          >
            <MenuItem value={0}>Engenharias</MenuItem>
            <MenuItem value={1}>Engenharia de Software</MenuItem>
            <MenuItem value={2}>Engenharia Aeroespacial</MenuItem>
            <MenuItem value={3}>Engenharia Eletr√¥nica</MenuItem>
            <MenuItem value={4}>Engenharia Automotiva</MenuItem>
            <MenuItem value={5}>Engenharia de Energia</MenuItem>        
          </Select>
        </FormControl>
      </form>
    );
  }
}

SimpleSelect.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(SimpleSelect);