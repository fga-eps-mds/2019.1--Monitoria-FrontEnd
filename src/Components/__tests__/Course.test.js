import React from 'react';
import Course from '../EditProfile/Course';
import { shallow } from 'enzyme';


describe('Testing Course component', () => {
  it('Test if card renders correctly', () =>{
      const tree = shallow(<Course/>);
      expect(tree).toMatchSnapshot();
  });
});