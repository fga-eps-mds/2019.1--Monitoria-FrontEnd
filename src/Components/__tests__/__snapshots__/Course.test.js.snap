// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders a snapshot properly 1`] = `
<form
  autoComplete="off"
  className="SimpleSelect-root-1"
>
  <div
    className="MuiFormControl-root-4 SimpleSelect-formControl-2"
  >
    <label
      className="MuiFormLabel-root-19 MuiInputLabel-root-8 MuiInputLabel-formControl-13 MuiInputLabel-animated-16 MuiInputLabel-outlined-18"
      data-shrink={false}
      htmlFor="outlined-age-simple"
    >
      Curso
    </label>
    <div
      className="MuiInputBase-root-46 MuiOutlinedInput-root-33 MuiInputBase-formControl-47"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="MuiPrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline-40"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="MuiPrivateNotchedOutline-legend-64"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <div
        className="MuiSelect-root-26"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          className="MuiSelect-select-27 MuiSelect-selectMenu-30 MuiSelect-outlined-29 MuiInputBase-input-56 MuiOutlinedInput-input-41"
          id="select-age"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          id="outlined-age-simple"
          name="age"
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-65 MuiSelect-icon-32"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</form>
`;

exports[`Testing Course component Test if Course renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(SimpleSelect) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "formControl": "SimpleSelect-formControl-2",
        "root": "SimpleSelect-root-1",
        "selectEmpty": "SimpleSelect-selectEmpty-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "formControl": "SimpleSelect-formControl-2",
          "root": "SimpleSelect-root-1",
          "selectEmpty": "SimpleSelect-selectEmpty-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing Course component Test if card renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(SimpleSelect) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "formControl": "SimpleSelect-formControl-2",
        "root": "SimpleSelect-root-1",
        "selectEmpty": "SimpleSelect-selectEmpty-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "formControl": "SimpleSelect-formControl-2",
          "root": "SimpleSelect-root-1",
          "selectEmpty": "SimpleSelect-selectEmpty-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
